postgresql/resource_postgresql_grant.go

```diff
@@ -450,7 +450,7 @@ func readRolePrivileges(txn *sql.Tx, d *schema.ResourceData) error {
 
 	case "function", "procedure", "routine":
 		query = `
-SELECT pg_proc.proname, array_remove(array_agg(privilege_type), NULL)
+SELECT CONCAT(pg_proc.proname, '(', pg_get_function_identity_arguments(pg_proc.oid), ')'), array_remove(array_agg(privilege_type), NULL)
```

Comment:
Thankfully postgres provides this function to get the exact argument list that's needed in a grant statement.